### ISSUE TITLE
Wrong subscribed since date for all subscriptions under My Subscriptions

### ISSUE THREAD

**Issue by azizLIGHT** [streamlink/streamlink-twitch-gui#627]
## Bug report

- [x] I understand the difference between *Streamlink* and *Streamlink Twitch GUI*.
- [x] This is a bug report and I have read the contribution guidelines.

### Environment details

**Operating system and version:**  
Windows 10

**Streamlink Twitch GUI version:**  
1.6.0

**Streamlink version:**  
v0.14.2 - Git 215d789

**Configuration details:**  
N/A

### Description
Subscription start date for every subscribed and gifted channel under My Subscriptions is wrong. The tooltip is also wrong. Both things are showing the timestamp of when Streamlink Twitch GUI was launched and as a result show "Subscribed for 0 months" for all channels, gifted and paid.

### Expected / Actual behavior

1. The subscription date should show actual date of subscription (or gift) for each subscribed channel not 0 months
2. The tooltip should show the exact date of subscription/gift start, not start date of Streamlink Twitch GUI

### Reproduction steps

Prerequisites
1. Already subscribed and gifted some channel subscriptions on Twitch for some time
3. Already logged in to Streamlink Twitch GUI

Reproduce
1. Open Streamlink Twitch GUI and note time of launch
2. Under 'My' on left panel, click Subscriptions
3. All the subscriptions listed show "Subscribed for 0 months"
4. Hover mouse over "Subscribed for 0 months" and look at tooltip
5. Tooltip will show timestamp of when you launched Streamlink Twitch GUI

### Log output

N/A

### Additional comments, screenshots, etc.

- There is no distinction between a gift and an actual subscription in this section either

**Comment by bastimeyer** [comment]
I can't reproduce this with the data I have here (a single Twitch Prime subscription). I would appreciate it if you or somebody else with the same issue could provide their subscription data here. This data can only be retrieved from Twitch's private API, so it's not documented and can change without them needing to deprecate things first.

This is how you can get it with `curl` and optionally format it with `jq`.

```bash
curl -s \
  -H 'Accept: application/vnd.twitchtv.v5+json' \
  -H 'Client-ID: phiay4sq36lfv9zu7cbqwz2ndnesfd8' \
  -H 'Authorization: OAuth YOUR_OAUTH_TOKEN_HERE' \
  'https://api.twitch.tv/api/users/YOUR_USER_NAME_HERE/tickets' \
  | jq
```

The API response looks like that:
https://github.com/streamlink/streamlink-twitch-gui/blob/v1.6.0/src/test/fixtures/data/models/twitch/ticket.json#L7-L52

And if you intend to post it here, make sure to replace the IDs and other sensitive data (if you care).

----

> 1. The subscription date should show actual date of subscription (or gift) for each subscribed channel not 0 months

This is how the subscription's month count gets calculated:

- https://github.com/streamlink/streamlink-twitch-gui/blob/v1.6.0/src/app/ui/components/list/subscription-item/component.js#L50-L64

As you can see, it'll try to use the `purchase_profile.consecutive_months` value if it exists, otherwise it'll calculate a time difference in months from the current time of your system and the subscription's `created_at` value.

> 2\. The tooltip should show the exact date of subscription/gift start, not start date of Streamlink Twitch GUI

This is not possible. The tooltip shows the date of the subscription's `created_at` property and formats it according to your locale settings:

- https://github.com/streamlink/streamlink-twitch-gui/blob/v1.6.0/src/app/ui/components/list/subscription-item/template.hbs#L41
- https://github.com/streamlink/streamlink-twitch-gui/blob/v1.6.0/src/app/ui/components/helper/format-time.js

Maybe this property is missing from the API response and it's using the current time instead. As I've said, I can't reproduce this with the data of my subscription.

**Comment by azizLIGHT** [comment]
@bastimeyer I'm trying to get this info but I do not know how to deal with error 401 Unauthorized: missing required oauth scope. 

I tried scope: null, and scope: channel_editor based off google searches, but they both give 401. What do I need here to give you that info

**Comment by bastimeyer** [comment]
You can get your OAuth token from the Twitch GUI if you enable advanced settings in the main settings menu. Then go to the user menu at the top and click the "show access token" button.

**Comment by azizLIGHT** [comment]
You misunderstand me. I have my oauth token already. What I don't have is the oauth scope. Either the API has changed, because it is as you said, undocumented, or I have the wrong scope, and I tried the ones I listed. So if you could guide me on that, I can get you the data you need

**Comment by bastimeyer** [comment]
The token is tied to the requested permissions scope and client ID. If you're using the access token from your login session in the Twitch GUI and also its client ID (which I've already included above), then you're automatically using the correct permissions and don't have to worry about that.

**Comment by azizLIGHT** [comment]
Got it, let me try again

**Comment by azizLIGHT** [comment]
I pay for 4 subs: forsen, nymn, sodapoppin, and virgerl. 
I got gifted 3 subs: becca, trainwreckstv, and sweet_anita (even if the api response says is_gift false)
[Heres the output](https://gist.github.com/azizLIGHT/986a18e2e0109b7439f7b0b83fb0f6cc)

**Comment by bastimeyer** [comment]
Thanks for the data, @azizLIGHT! I will take a look at it when I get the time.
